### DISCUSSION TITLE
How to set default value in scientific notation?

### DISCUSSION THREAD

**Discussion by ybyygu** [mitsuhiko/minijinja#216] [Q&A] [answered]
I want to render a variable with a default value in scientific notation like this:
> EDIFF={{EDIFF or 1E-6}}

but failed with following error

![image](https://user-images.githubusercontent.com/128990/223420962-b37d6ae3-fbf9-49f8-a67c-022ab0552f22.png)

This works
> EDIFF={{EDIFF or 1/10**6}}

output

> EDIFF=1e-6

Is this a bug? Is there other way to set default value in scientific notation? Thanks in advance!

**Comment by mitsuhiko** [comment]
Scientific notation is not yet implemented. Happy to accept a pull request, support for that syntax exists in Jinja2.

**Comment by ybyygu** [comment, marked as answer]
I fixed it in #218. I hope this is acceptable to you.

**Reply by mitsuhiko** [reply]
Thank you. I merged it with a small refactor!
